### ISSUE TITLE
SmartFileSystem microSD card does not boot

### ISSUE THREAD

**Issue by AmiKit** [michalsc/Emu68#219]
A microSD card formatted with SmartFileSystem does not boot. Interestingly enough, if my SmartFileSystem formatted CF card is plugged in, with lower boot priority, the microSD card boots fine. But never if plugged alone, without the CF card.

See the attached screenshots how my SFS boot partition is set up (I also tried SFS 1.280) but with the same results. Detailed instructions are here: https://docs.google.com/document/d/18TSFPV62gznqmiSmbm9ZKSbfzW-o1iWrUExNbU8dyb4/edit#heading=h.vdd4yhlab0ty

Thank you.
![sfs1](https://github.com/michalsc/Emu68/assets/131966133/b5c040bb-8353-4442-9f18-f511e508973c)
![sfs2](https://github.com/michalsc/Emu68/assets/131966133/86556e2e-9549-43e6-befc-9d5b36dbf8fb)
![sfs3](https://github.com/michalsc/Emu68/assets/131966133/66afe395-c20c-4615-bdda-a041d91ce70a)

**Comment by merlinkv** [comment]
I don't use FFS or SFS on any Amiga, emulated or real, only the PFS3AIO 19.2, do you have tried with PFS3AIO?

Mi Pistorm32 Lite Emu68 is working fine with mSD/32GB and four partitions.

**Comment by AmiKit** [comment]
> I don't use FFS or SFS on any Amiga, emulated or real, only the PFS3AIO 19.2, do you have tried with PFS3AIO?

Yes, PFS3AIO boots fine with the very same microSD card. This bug is about SFS only, it seems.

**Comment by AmiKit** [comment]
A new beta of PFS3AIO 3.2 (0.20) appeared and it seems that as soon as I update the standard PFS 3.1 (0.19) formatted partition with PFS 3.2 (0.20), it doesn't boot anymore under Emu68, while it still boots under WinUAE.

**Comment by AmiKit** [comment]
One more interesting finding. Consider this scenario:
AK0: is a bootable partition with PFS 3.1, blocksize 512.
AK1: is a non-bootable partition with SFS 1.280, blocksize 1024
The card boot under latest Emu68.

BUT as soon as I increase the AK1: non-bootable partition blocksize to 4096 and format the partition, the card does not boot anymore... strange, isn't it?

**Comment by craziazkowboi** [comment]
This issue extends to PFS 19.3 as well.  I am attempting a new build and seeing similar issues:

At first I thought I had a hardware issue, this was not the case,  I have run buptest successfully in Emu68 and using 9 iterations, as well as both gpiotest and buptest in Musashi.  Musashi starts without a hitch.  

Only Emu68 and PFS3AIO (v19.3) fail, If I use FFS Emu68 works, but as soon as I add any partition as PFS3AIO it fails (I tried adding an FFS partition as my first and boot partition, and then add a PFS as DH1.  As soon as  I reboot with any PFS partition, Emu68 no longer boots to the kickstart screen (requesting a floppy to be inserted, also no clicks from drives).  

From here I need to run diskpart on a windoze PC again, clean the card, recreate my partitions, and boot AmigaOS 3.2 from Floppy again.

I have ensured and triple checked that my settings in HDToolbox are saved, including the mask.

I have followed both Michael’s guide, and the retro32 guide for setting up a new SDCard for Emu68.  My SDCard is good, and I have also tried another SDCard with the same results.  Both work with Musashi, or Emu68 with FFS.  Neither SDCard work with PFS and Emu68.

**Comment by TuKo1982** [comment]
I can confirm that updating to pfs3aio 20.0 test5 makes my image unbootable (blocksize untouched from 19.2). 

Downgrading back to 19.2 fixes it.

**Comment by codewiz** [comment]
I think I've hit this bug too. Steps to reproduce:

1. Add L:SmartFilesystem 1.279 to the RDB, with DosType 0x53465300
2. Create a partition and assign it to SFS\00
3. Mark the partition "Automount". This is the default, and it is required to trigger the bug.
4. Reboot: the system will hang after the Emu68 splash, but before the Amiga Early Startup screen.

NOTE: the same SmartFilesystem binary works in Amiberry with 68040 CPU emulation and also in a real Amiga 4000/040.

<img width="656" height="222" alt="Image" src="https://github.com/user-attachments/assets/e3a8d964-1456-4cc4-82a4-a57a3fe96d2a" />

**Comment by codewiz** [comment]
Proposed fix: https://github.com/michalsc/brcm-sdhc.device/pull/1

Tested with SmartFileSystem 1.279 on a Pistorm32-lite. Mounts 4 SFS patitions and boots off one of them.

Sorry, the actual fix for this bug is mixed with several other changes I did while debugging blindly.  I'm happy to remove some of them or split them to separate  commits.

**Comment by codewiz** [comment]
The fix landed in brcm-sdhc.device (not yet released in Emu68):
  https://github.com/michalsc/brcm-sdhc.device/commit/8b0be38c75a905836cb23e8a5132670b0a771627

**Comment by michalsc** [comment]
Closing since fixed by @codewiz
